docs/Guides/Database.md

```diff
@@ -0,0 +1,307 @@
+})
+
+fastify.listen(3000, err => {
+  if (err) throw err
+  console.log(`server listening on ${fastify.server.address().port}`)
+})
+```
+
+### Writing plugin for a database library
+We could write a plugin for a database 
+library too (e.g Knex, Prisma, TypeORM, etc.). 
+We will use [Knex](https://knexjs.org/) in our example.
+
+```javascript
+'use strict'
+
+const fp = require('fastify-plugin')
+const knex = require('knex')
+
+function knexPlugin(fastify, options, done) {
+  if(!fastify.knex) {
+    const knex = knex(options)
+    fastify.decorate('knex', knex)
+
+    fastify.addHook('onClose', (fastify, done) => {
+      if (fastify.knex === knex) {
+        fastify.knex.destroy(done)
+      }
+    })
+  }
+
+  done()
+}
+
+export default fp(plugin, { name: 'fastify-knex-example' })
+```
+
+### Writing a plugin for a database engine
+
+In this example, we will create a basic Fastify MySQL plugin 
+from scratch (it is a stripped-down example, please use the official plugin in production).
+
+```javascript
+const fp = require('fp')
+const mysql = require('mysql2/promise')
+
+function fastifyMysql(fastify, options, done) {
+  const connection = mysql.createConnection(options)
+
+  if (!fastify.mysql) {
+    fastify.decorate('mysql', connection)
+  }
+
+  fastify.addHook('onClose', (fastify, done) => connection.end().then(done).catch(done))
+
+  done()
+}
+
+export default fp(fastifyMysql, { name: 'fastify-mysql-example' })
+```
+
+### Migrations
+
+Database schema migrations are an integral part of database management and development. Migrations provide repeatable and testable way to modify database's schema and to prevent data loss.
+
+As stated in the beginning of the guide, Fastify is database agnostic and any NodeJS database migration tool can be used with it. We will give an example of using [Postgrator](https://www.npmjs.com/package/postgrator) which has support for Postgres, MySQL and SQL Server. For MongoDB migrations, please check [migrate-mongo](https://www.npmjs.com/package/migrate-mongo).
+
+#### [Postgrator](https://www.npmjs.com/package/postgrator)
+
+Postgrator is NodeJS SQL migration tool that uses a directory of SQL scripts to alter the database schema. Each file an migrations folder need to follow the pattern: ` [version].[action].[optional-description].sql`.
+
+**version:** must be an incrementing number (e.g. `001` or a timestamp).
+
+**action:** should be `do` or `undo`. `do` implements the version, `undo` reverts it. Think about it like `up` and `down` in other migration tools.
+
+**optional-description** describes which changes migration makes. Altough optional, it should be used for all migrations as it makes it easier for everyone to know which changes are made in a migration.
+
+In our example we are going to have a single migration that creates a `users` table and we are going to use `Postgrator` to run the migration.
+
+> Run `npm install pg postgrator` to install dependencies needed for the example.
```

Comment:
I haven't had time to add a CI check for this, but please wrap lines to ~80 characters. If using VSCode to edit this file, you can use https://marketplace.visualstudio.com/items?itemName=stkb.rewrap

Reply:
Wrapped, thanks!